src/ast/ast_build_filter_tree.c

```diff
@@ -392,6 +350,12 @@ FT_FilterNode *AST_BuildFilterTreeFromClauses(const AST *ast,
 		if(predicate) AST_ConvertFilters(&filter_tree, predicate);
 	}
 
+	if(!FilterTree_Valid(filter_tree)) {
+		// Invalid filter tree structure, a compile-time error has been set.
+		FilterTree_Free(filter_tree);
+		return NULL;
+	}
```

Comment:
Do we want / need to perform this test in this stage?

Reply:
We do, this should have been added when adding the `AST_BuildFilterTreeFromClauses` function. All of the added WHERE-erroring tests are triggered here.